src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -267,9 +268,9 @@ public override SourceText WithChanges(IEnumerable<TextChange> changes)
             private class ChangedSourceText : SnapshotSourceText
             {
                 private readonly SnapshotSourceText _baseText;
-                private readonly ITextSnapshot _baseSnapshot;
+                private readonly ITextImage _baseSnapshot;
```

Comment:
I'm fairly certain this code is pretty terrible.  However, i'm going to leave things as is so that we don't subtly break anything in 15.3